app/src/main/kotlin/org/eclipse/kuksa/companion/MainActivity.kt

```diff
@@ -286,6 +301,12 @@ class MainActivity : ComponentActivity() {
     }
 
     private fun connectToDataBroker(onConnected: () -> Unit = {}) {
+        // dataBrokerConnection is already established e.g. after an orientation change
+        if (dataBrokerConnection != null) {
+            onConnected()
```

Comment:
Our Listeneners (vssDoorListener, vssTemperatureListener, vssLightListener, ...) are stored inside the MainActivity, which is destroyed due to the orientation change. In onDestroy we cleanup by removing all subscriptions (unsubscribe) again, which is fine from a lifecycle point of view (subscribe onConnected() <-> unsubscribe onDestroy) to prevent leaking of any ressources. 

> Is the onConnected() correct here?

Yes, the new "onConnected()" call in connectToDataBroker is required to re-register the listeners and bring the MainActivity back into a working state. 

> Doesn't it lead to duplicate subscriptions?

No, since we cleanup in onDestroy the old subscriptions. 

To fix this issue we would need to elevate the logic from the MainActivity into the application, or probably better (?) into the viewModels and remove the "unsubscribe" call from MainActivity#onDestroy.

We should test this thoroughly and also check all other properties in MainActivity if they are stateful or not. As you suggested, let's do this in another ticket.

Reply:
Created #55